File: 739-daily-temperatures/daily-temperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        Deque<Integer> stack = new ArrayDeque<>();
        int n=temperatures.length;
        int i=0;
        while(i<n){
            while(!stack.isEmpty() && temperatures[i]>temperatures[stack.peekLast()]){
                temperatures[stack.peekLast()] = i-stack.removeLast();
            }
            stack.addLast(i);
            i++;
        }
        while(!stack.isEmpty()){
            temperatures[stack.removeLast()]=0;
        }
        return temperatures;
    }
}

File: 739-daily-temperatures/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        st=[]
        st.append(0)
        for i in range(1, len(temperatures)):
            while len(st)>0  and temperatures[st[-1]]<temperatures[i]:
                x=st.pop()
                temperatures[x] = i-x
            st.append(i)
        i=len(temperatures)-1
        while len(st)>0:
            temperatures[st.pop()]=0
            i -= 1
        return temperatures
        